### PR TITLE
Finish migration of QPE kata and publish it

### PR DESCRIPTION
**Pull request by tcNickolas**

### CODE REVIEW

File: katas/content/phase_estimation/index.md [hunk 1]
```diff
@@ -102,17 +102,151 @@ $$\theta = 0.\theta_1 \theta_2... \theta_n = \frac{\theta_1}{2^1}+ \frac{\theta_
...
 @[section]({
     "id": "phase_estimation__qpe",
     "title": "Quantum Phase Estimation Algorithm"
 })
 
-- theory
-- exercise: task 1.4 to implement QPE
-- demo of end-to-end probabilistic behavior in case of lower precision (use R1 gate)
+Quantum phase estimation algorithm is the most common algorithm used for estimating the eigenphase that corresponds to the given unitary-eigenvector pair.
+It relies on quantum Fourier transform, and allows you to estimate $n$ binary digits of the eigenphase $\theta$.
```
Comment (cesarzc) [inline katas/content/phase_estimation/index.md:118]:
```suggestion
The quantum phase estimation algorithm is the most common algorithm used for estimating the eigenphase that corresponds to the given unitary-eigenvector pair.
It relies on the quantum Fourier transform, and allows you to estimate $n$ binary digits of the eigenphase $\theta$.
```

File: katas/content/phase_estimation/index.md [hunk 2]
```diff
@@ -102,17 +102,151 @@ $$\theta = 0.\theta_1 \theta_2... \theta_n = \frac{\theta_1}{2^1}+ \frac{\theta_
...
+What does this sequence of gates do?
+
+Let's consider the scenario in which the first register is in the basis state $\ket{k} = \ket{k_0} \otimes \ket{k_1} \otimes ... \otimes \ket{k_{n-1}}$, where $k$ is an integer with $n$ bits in its binary notation: 
+$$k = 2^0k_0 + 2^1 k_1 + 2^2k_2 + ... + 2^{n-1}k_{n-1}$$
+
+In this case, this sequence of gates will apply the unitary $U^k$ to the second register! Indeed, 
+
+1. Controlled $U$ gate with the first qubit as control will apply $U$ gate if $k_0 = 1$, and do nothing if $k_0 = 0$.
+2. Controlled $U^2$ gate with the second qubit as control will apply $U^2$ gate if $k_1 = 1$, and do nothing if $k_1 = 0$.
+3. Controlled $U^4$ gate with the third qubit as control will apply $U^4$ gate if $k_2 = 1$, and do nothing if $k_2 = 0$.
```
Comment (cesarzc) [inline katas/content/phase_estimation/index.md:162]:
```suggestion
1. Controlled $U$ gate with the first qubit as control will apply the $U$ gate if $k_0 = 1$, and do nothing if $k_0 = 0$.
2. Controlled $U^2$ gate with the second qubit as control will apply the $U^2$ gate if $k_1 = 1$, and do nothing if $k_1 = 0$.
3. Controlled $U^4$ gate with the third qubit as control will apply the $U^4$ gate if $k_2 = 1$, and do nothing if $k_2 = 0$.
```

File: katas/content/phase_estimation/index.md [hunk 3]
```diff
@@ -102,17 +102,151 @@ $$\theta = 0.\theta_1 \theta_2... \theta_n = \frac{\theta_1}{2^1}+ \frac{\theta_
...
+2. An $m$-qubit state $\ket{\psi}$ - an eigenvector of $U$.
+3. An integer $n$ that specifies the desired binary precision of the eigenphase estimate.
+
+### The starting state
+
+The algorithm acts on $n+m$ qubits, split in two registers.
+- The first $n$ qubits start in the state $\ket{0...0} = \ket{0}^{\otimes N}$. These qubits are used to get the eigenvalue.
+- The last $m$ qubits start in the state $\ket{\psi}$ - the eigenvector of $U$.
+
+### Step 1. Apply Hadamard transform to each qubit of the first register
```
Comment (cesarzc) [inline katas/content/phase_estimation/index.md:133]:
```suggestion
### Step 1. Apply the Hadamard transform to each qubit of the first register
```

File: katas/content/phase_estimation/index.md [hunk 4]
```diff
@@ -102,17 +102,151 @@ $$\theta = 0.\theta_1 \theta_2... \theta_n = \frac{\theta_1}{2^1}+ \frac{\theta_
...
+$$U^k\ket{\psi} = (e^{2 \pi i \theta})^k \ket{\psi} = e^{2 \pi i \theta \cdot k} \ket{\psi}$$
+
+You can use this to write the following expression for the state of the system after this step:
+
+$$\frac1{\sqrt{2^n}}\sum_{k=0}^{2^n-1} e^{2 \pi i \theta \cdot k} \ket{k} \otimes \ket{\psi}$$
+
+Notice that the first and second registers end up not being entangled with each other.
+
+
+### Step 3. Apply inverse QFT
```
Comment (cesarzc) [inline katas/content/phase_estimation/index.md:184]:
```suggestion
### Step 3. Apply the inverse QFT
```

File: katas/content/phase_estimation/index.md [hunk 5]
```diff
@@ -102,17 +102,151 @@ $$\theta = 0.\theta_1 \theta_2... \theta_n = \frac{\theta_1}{2^1}+ \frac{\theta_
...
+
+Now, recall that quantum Fourier transform that acts on a basis state $\ket{j}$ that represents an $n$-bit integer $j$ performs the following transformation:
```
Comment (cesarzc) [inline katas/content/phase_estimation/index.md:186]:
```suggestion
Now, recall that the quantum Fourier transform that acts on a basis state $\ket{j}$ that represents an $n$-bit integer $j$ performs the following transformation:
```

File: katas/content/phase_estimation/index.md [hunk 6]
```diff
@@ -102,17 +102,151 @@ $$\theta = 0.\theta_1 \theta_2... \theta_n = \frac{\theta_1}{2^1}+ \frac{\theta_
...
+
+$$\ket{j} \rightarrow \frac1{\sqrt{2^n}}\sum_{k=0}^{2^n-1}  e^{2\pi i jk/2^n} \ket{k} = \frac1{\sqrt{2^n}}\sum_{k=0}^{2^n-1}  e^{2\pi i (j/2^n) \cdot k} \ket{k}$$
+
+If you compare this expression with the state of the first register at the end of the previous step, you'll notice that they match exactly, with $\theta = j / {2^n}$.
+
+This means that you can use inverse Fourier transform (the adjoint of the Fourier transform) to convert the state of the first register to a binary notation of the integer that stores the first $n$ bits of $\theta$ written as a binary fraction.
+If $\theta = 0.\theta_1 \theta_2... \theta_n$, applying inverse QFT to the first register will yield a state
```
Comment (cesarzc) [inline katas/content/phase_estimation/index.md:193]:
```suggestion
If $\theta = 0.\theta_1 \theta_2... \theta_n$, applying the inverse QFT to the first register will yield a state 
```

**Review by cesarzc — APPROVED** [review]

File: katas/content/phase_estimation/implement_qpe/Solution.qs [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+        P : Qubit => Unit,
+        n : Int
+    ) : Int {
+        use (phaseRegister, eigenstate) = (Qubit[n], Qubit());
+        P(eigenstate);
+        ApplyToEach(H, phaseRegister);
+        for k in 0 .. n - 1 {
+            for _ in 1 .. 1 <<< k {
+                Controlled U([phaseRegister[k]], eigenstate);
+            }
```
Comment (swernli) [inline katas/content/phase_estimation/implement_qpe/Solution.qs:13]: nitpick: trailing spaces

```suggestion
            }
```

File: katas/content/phase_estimation/one_bit_eigenphase/Solution.qs [hunk 8]
```diff
@@ -0,0 +1,11 @@
+namespace Kata {
+    operation OneBitPhaseEstimation(U : Qubit => Unit is Ctl, P : Qubit => Unit) : Int {
+        use (control, eigenstate) = (Qubit(), Qubit());
+        H(control);
```
Comment (swernli) [inline katas/content/phase_estimation/one_bit_eigenphase/Solution.qs:4]: nitpick: trailing spaces

```suggestion
        H(control);
```

**Review by swernli — APPROVED** [review]

**Pull request merged**